.github/workflows/deploy-to-control-plane.yml

```
@@ -34,60 +36,100 @@
-          ref: ${{ github.event_name == 'pull_request' && github.event.pull_request.head.ref || steps.getRef.outputs.PR_REF || github.ref }}
 
-      - name: Validate Required Secrets
+      - name: Setup Environment
+        uses: ./.github/actions/setup-environment
+
+      - name: Get PR Number for Push Event
+        if: github.event_name == 'push'
+        id: get-pr
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2086:info:4:34: Double quote to prevent globbing and word splitting

(shellcheck)